crates/starknet-server/tests/test_trace.rs

```diff
@@ -189,4 +188,54 @@ mod trace_tests {
             panic!("Could not unpack the transaction trace from {deploy_account_tx_trace:?}");
         }
     }
+
+    #[tokio::test]
+    async fn get_traces_from_block() {
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+
+        let mint_tx_hash: FieldElement = devnet.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+
+        // Currently, we support only one transaction per block, so if this changes in the future
+        // this test also needs to be updated
+        let block_traces = &devnet
+            .send_custom_rpc("starknet_traceBlockTransactions", json!({ "block_id": "latest" }))
+            .await["result"];
+
+        // assert transaction hash
+        assert_eq!(
+            mint_tx_hash,
+            FieldElement::from_hex_be(
+                block_traces["traces"][0]["transaction_hash"].as_str().unwrap()
```

Comment:
Is len of `block_traces["traces"]` supposed to be one? If so, I would add a length assertion before all the assertions, and extract `block_traces["traces"][0]` into a `trace` variable to reduce duplication.

Reply:
addressed in cdfa97f922e9acc47385258a71ad42a3195bfae9

Reply:
If you already deserialize it with
```
serde_json::from_value::<Vec<BlockTransactionTrace>>(block_traces.clone(),)
```
why not later use `validate_invocation`, `execute_invocation` and `fee_transfer_invocation` of that deserialized object that is only used for length assertion?

Also, no need to call the variable `traces` if it's just one `trace` (the one at index 0)

Reply:
that would require if let type match which would increase code size

Reply:
I see. but why can't the length be asserted on the object whose property [0] you are taking? I.e. without deserialization?

Reply:
because it's `&Value` type

Reply:
ok then